lib/exchange/simpleswap/simpleswap_exchange_provider.dart

```diff
@@ -0,0 +1,217 @@
+import 'dart:convert';
+
+import 'package:cake_wallet/exchange/exchange_pair.dart';
+import 'package:cake_wallet/exchange/exchange_provider.dart';
+import 'package:cake_wallet/exchange/exchange_provider_description.dart';
+import 'package:cake_wallet/exchange/simpleswap/simpleswap_request.dart';
+import 'package:cake_wallet/exchange/trade_not_created_exeption.dart';
+import 'package:cake_wallet/exchange/trade_not_found_exeption.dart';
+import 'package:cake_wallet/exchange/trade_state.dart';
+import 'package:cw_core/crypto_currency.dart';
+import 'package:cake_wallet/exchange/trade_request.dart';
+import 'package:cake_wallet/exchange/trade.dart';
+import 'package:cake_wallet/exchange/limits.dart';
+import 'package:cake_wallet/.secrets.g.dart' as secrets;
+import 'package:http/http.dart';
+
+class SimpleSwapExchangeProvider extends ExchangeProvider {
+  SimpleSwapExchangeProvider()
+      : super(
+            pairList: CryptoCurrency.all
+                .map((i) =>
+                    CryptoCurrency.all.map((k) => ExchangePair(from: i, to: k, reverse: true)).where((c) => c != null))
+                .expand((i) => i)
+                .toList());
+
+  static const apiAuthority = 'api.simpleswap.io';
+  static const getEstimatePath = '/v1/get_estimated';
+  static const rangePath = '/v1/get_ranges';
+  static const getExchangePath = '/v1/get_exchange';
+  static const createExchangePath = '/v1/create_exchange';
+  static const apiKey = secrets.simpleSwapApiKey;
+
+  @override
+  ExchangeProviderDescription get description =>
+      ExchangeProviderDescription.simpleSwap;
+      
+  @override
+  Future<double> calculateAmount(
+      {CryptoCurrency from, CryptoCurrency to, double amount, bool isFixedRateMode, bool isReceiveAmount}) async {
+    try {
+      if (amount == 0) {
+        return 0.0;
+      }
+      final fromCurrency = _normalizeCryptoCurrency(from);
+      final toCurrency = _normalizeCryptoCurrency(to);
+      final params = <String, String>{
+        'api_key': apiKey,
+        'currency_from': fromCurrency,
+        'currency_to': toCurrency,
+        'amount': amount.toString(),
+        'fixed': isFixedRateMode.toString()
+      };
+      final uri = Uri.https(apiAuthority, getEstimatePath, params);
+      
+      final response = await get(uri);
+
+      if (response.body == null) return 0.00;
+      final responseJson = json.decode(response.body) as String;
+    
+      return double.parse(responseJson);
```

Comment:
if there should be Json, how you could to parse double from that ? I think here is wrong name for `responseJson` variable.